### ISSUE TITLE
Navigation Testing

### ISSUE THREAD

**Issue by clintwood** [EmilTholin/svelte-routing#104]
I'm having an issues with testing navigation using `Jest` +  `@testing-library/svelte`.

Essentially the issues I'm having are:
1) the navigation state (i.e. the `ROUTER` context) seems to persist between tests
2) if I navigate from one route to another then use `window.history.back()` navigation does not go back.

I realise this is a bit general and point 2 _may_ not be directly related to `svelte-routing` but if anyone has been able to successfully test navigation using `Jest` +  `@testing-library/svelte` I'd be interested to hear how you are doing this.

**Comment by clintwood** [comment]
I think I've resolved the issue for point 2 - I think it was an async timing issue since internally jest fires the `popstate` event using `setTimeout` which only runs at some later time so I'm doing the following in my tests which seems to work:

```js
...
    await fireEvent.click(elt); // on:click={() => window.history.back()}
    await wait(() => {
      elt = getByTestId('root');
      expect(elt).toHaveTextContent('ROOT');
    });
...

For point 1 above I need to find a way to clear the `ROUTER` context on each test!
```

**Comment by mefechoel** [comment]
I think this is an issue with jsdom, which `@testing-library/svelte` is using internally. Its history API implementation is not complete. I ran into the same issue when testing my own routing library, which is actually based on svelte-routing.
I ended up using cypress, which runs in a real browser, so it works just fine. There is also a [svelte package](https://github.com/bahmutov/cypress-svelte-unit-test) for cypress, which might help you. I have however not used it myself.
